pybit/__init__.py

```diff
@@ -1131,43 +1133,31 @@ def close_position(self, symbol):
         """
 
         # First we fetch the user's position.
-        r = self._submit_request(
-            method='GET',
-            path=self.endpoint + '/v2/private/position/list',
-            query={'symbol': symbol},
-            auth=True
-        )
-
-        # Next we detect the position size and side.
         try:
-            position = r['result']['size']
-            side = 'Buy' if r['result']['side'] == 'Sell' else 'Sell'
-            if position == 0:
-                return self.logger.error('No position detected.')
+            r = self.my_position(symbol=symbol)['result']
 
         # If there is no returned position, we want to handle that.
-        except IndexError:
+        except KeyError:
             return self.logger.error('No position detected.')
 
-        # We should know if we're pinging the inverse or linear perp API.
-        if symbol.endswith('USDT'):
-            close_path = self.endpoint + '/private/linear/order/create'
-        else:
-            close_path = self.endpoint + '/v2/private/order/create'
-
-        # Submit a market order against your position for the same qty.
-        return self._submit_request(
-            method='POST',
-            path=close_path,
-            query={
+        # Next we generate a list of market orders
+        orders = [
+            {
                 'symbol': symbol,
                 'order_type': 'Market',
-                'side': side,
-                'qty': position,
-                'time_in_force': 'ImmediateOrCancel'
-            },
-            auth=True
-        )
+                'side': 'Buy' if p['side'] == 'Sell' else 'Sell',
```

Comment:
You've defined `r` to be the result of `my_position()`. Did you mean to use `r` rather than `p`?

Reply:
Ah nvm I see what you're doing. Is there a reason why to generate multiple orders to close a position rather than a single one?

Reply:
Hi mate, linear (USDT) contracts can have multiple positions opened at once (Buy+Sell sides), Bybit returns both side positions in a list of dicts for USDT when calling my_position(), as opposed to a single dict (no list) for non-linear, so these changes make the function universal to cover both scenarios via the list comprehension and use of isinstance to check if a list is returned

Reply:
Ah I wasn't aware of the multiple positions on linear. I should use it more often 😛 .

Beautifully done sir!